### ISSUE TITLE
Support Kafka headers in Kafka Source

### ISSUE THREAD

**Issue by daniilguit** [timberio/vector#6806]
Currently when defining Kafka Source I receive topic, partition, offset, message, but do not received kafka headers (string -> string dictionary) that can be present in any Kafka message.

Expecting to have something like this:
```
{
  "message" : "53.126.150.246 - - [01/Oct/2020:11:25:58 -0400] \"GET /disintermediate HTTP/2.0\" 401 20308",
  "offset" : 100,
  "partition" : "partition",
  "timestamp" : "2020-10-10T17:07:36+00:00",
  "topic" : "topic",
  "headers": {
    "MyCustomKafkaHeader": "Header Value"
  }
}
```

**Comment by jszwedko** [comment]
👍 it seems like we have easy access to these via the rdkafka crate so this shouldn't be too hard:

https://docs.rs/rdkafka/0.26.0/rdkafka/message/trait.Message.html#tymethod.headers

**Comment by jszwedko** [comment]
Resolved by #7030

**Comment by daniilguit** [comment]
@jszwedko: Cool! Thanks to all of you.
